src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
Sorry to labour the point here @lh3, but I just don't understand the arguments for doing this on the server side. It's not about whether a server _can_ do it efficiently, it's whether it _should_ be doing it at all. 

I'm not suggesting we do it client side using Python necessarily, I just used an example of a client library I'm familiar with. There's no reason this couldn't be done in C, if a C client library is written. I don't understand how this would effect server performance either --- servers are written to deal with many concurrent requests. Having a bunch of concurrent requests coming from a single client is no problem. Indeed, one could easily argue that it is better for the server to serve several simple queries concurrently than to need to serially process the equivalent complex query.

I guess the basic question I have is, why not entirely (1)? Since we must have client libraries in any case, why not push complex features that can be composed out of the atomic units the server provides into these libraries? This way, client libraries can rapidly iterate to add interesting and convenient features and the protocol can stay relatively static. When some of these features prove to be very useful and there is a compelling performance argument for moving them on to the server, we update the protocol.